### PR TITLE
Changed the default value for max_open_connections to one.

### PR DESCRIPTION
**Pull request by christ0pher**
If the value is Leq 0 an error over the python logging system ist emitted. This can be used to catch failures over network logging and custom defined handlers.

Connections are now checked only against the max_open property.

### CODE REVIEW

**Comment by methane** [comment]
Why should I change the default?
Why not use exception instead of logging?

**Comment by christ0pher** [comment]
Hi,

1. The default value should be a reasonable value. 0 for unlimited open connections looks more like a hack to me, instead of a reasonable value. Unlimited connections do not make sense in the context of databases (e.g file descriptor limit). If you use a pool a reasonable value is 1 to open at max one connection.

2. exceptions in constructors are not a nice way to show the error. Better way is to set the default value to a reasonable value. Also an unlimited amout of connections will crash the db if the load gets too high. 

best

**Comment by methane** [comment]
> The default value should be a reasonable value. 0 for unlimited open connections looks more like a hack to me, instead of a reasonable value. Unlimited connections do not make sense in the context of databases (e.g file descriptor limit). If you use a pool a reasonable value is 1 to open at max one connection.

I don't feel 1 is reasonable value.  It's too small for real application.

http://golang.org/pkg/database/sql/#DB.SetMaxOpenConns
Go's connection pool is unlimited by default.

> exceptions in constructors are not a nice way to show the error. 

I disagree.  Exception is better than logging.
Python's Zen saids:

>     Errors should never pass silently.

I really like it. If people misconfigured, libraries should crash ASAP.

**Comment by christ0pher** [comment]
Ok. i will change it to an exception.

But still initializing databasepools with unlimited connections is not the way to go. Even in GO.

**Comment by christ0pher** [comment]
Additional thoughts to the commit message: 
If you have a product and many services can open unlimited connection pools to the database, it is possible to bring down your hole product with just one service.
People who program should be aware what they are doing, but sometimes a little hint might be useful to build more stable systems.

File: tornado_mysql/pools.py [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 DEBUG = False
 
+POOLLOGGER = logging.getLogger("Tornado-MySQL")
```
Comment (methane) [inline tornado_mysql/pools.py]: remove this

File: tornado_mysql/pools.py [hunk 2]
```diff
@@ -47,7 +48,10 @@ def __init__(self,
         self.io_loop = io_loop or IOLoop.current()
         self.connect_kwargs = connect_kwargs
         self.max_idle = max_idle_connections
-        self.max_open = max_open_connections
+        if max_open_connections <= 0 or max_idle_connections <= 0:
+            raise ValueError("max idle or open connections not initialized properly")
+        else:
+            self.max_open = max_open_connections
```
Comment (methane) [inline tornado_mysql/pools.py]: `else:` should be removed.

File: tornado_mysql/pools.py [hunk 3]
```diff
@@ -47,7 +48,10 @@ def __init__(self,
         self.io_loop = io_loop or IOLoop.current()
         self.connect_kwargs = connect_kwargs
         self.max_idle = max_idle_connections
-        self.max_open = max_open_connections
+        if max_open_connections <= 0 or max_idle_connections <= 0:
+            raise ValueError("max idle or open connections not initialized properly")
```
Comment (methane) [inline tornado_mysql/pools.py]: `max_idle_connections=0` is valid.

**Comment by methane** [comment]
Our company use this module in production already.
Please don't break backward compatibility so easy.

**Pull request closed without merging**